assets/js/babyYoda.js

```diff
@@ -0,0 +1,47 @@
+
+const spawnBabyYoda = (randPosX, randPosY, colour) => {
+    const babyYoda = add([
+      rect(40, 40), // placeholder until we have a sprite
+      pos(randPosX, randPosY),
+      color(colour), // colour of the box until we have a sprite
+      "babyYoda", // tag to reference later on
+    ]);
+
+        //check input on each frame
+        babyYoda.onUpdate(() => {
+
+            // Define the keyboard keys for movement
+            const MOVE_LEFT_KEY = "left";
+            const MOVE_RIGHT_KEY = "right";
+            const MOVE_UP_KEY = "up";
+            const MOVE_DOWN_KEY = "down";
+
+            const movementSpeed = 1;
+
+            // Handle keyboard input
+            onKeyDown(MOVE_LEFT_KEY, () => {
+                if (babyYoda.pos.x > 0) {
+                    babyYoda.move(-movementSpeed, 0);
+
+                }
+                
+            });
+
+            onKeyDown(MOVE_RIGHT_KEY, () => {
+                // if (babyYoda.pos.y > 0) {
+                    babyYoda.move(movementSpeed, 0);
+                // }
```

Comment:
Suggestion: I think we could add a condition here to check if the position of baby Yoda is less than the width of the canvas, then they should be able to move right, otherwise it does nothing. That would prevent them from moving off the canvas. It would look something like this:
```
if(babyYoda.pos.x < width()){
  babyYoda.move(movementSpeed, 0);
}
```